core/src/main/java/org/apache/gravitino/storage/relational/session/SqlSessions.java

```diff
@@ -52,8 +62,8 @@ public static SqlSession getSqlSession() {
               .getSqlSessionFactory()
               .openSession(TransactionIsolationLevel.READ_COMMITTED);
       sessions.set(sqlSession);
-      return sqlSession;
     }
+    sessionCount.set(sessionCount.get() + 1);
```

Comment:
This is not atomic, you'd better use `AtomicInteger`.

Reply:
Thanks for the suggestion. While the ThreadLocal already ensures thread safety here, I agree that using `AtomicInteger` makes the intent of a counter more explicit and the code cleaner. I've made the change.